### DISCUSSION TITLE
Why doesn't Any report an incompatibility with more specific types?

### DISCUSSION THREAD

**Discussion by bjornmosten** [microsoft/pyright#7402] [Q&A] [answered]
See image:
![image](https://github.com/microsoft/pyright/assets/59401189/ae8691d5-a928-4262-a784-59781fb0d734)

I would expect a_client_0 to have the same result as b_client_0. What am I missing?

**Comment by erictraut** [comment, marked as answer]
The `Any` type is a special form used for "gradual typing" in the Python type system. You can read about it in the [official typing spec](https://typing.readthedocs.io/en/latest/spec/special-types.html#any). 

When assigning to a variable whose type is explicitly declared as `Any` (such as with `a_arg1`), pyright never narrows the type. 

By contrast, the variable `b_arg1` has no declared value, so its type is narrowed on assignment (in this case to `None`).

**Reply by bjornmosten** [reply]
Thank you
